mutex_flock.go

```diff
@@ -67,8 +68,45 @@ func acquireFlock(name string, done <-chan struct{}) <-chan acquireResult {
 	}
 
 	flockName := filepath.Join(os.TempDir(), "juju-"+name)
+	chownFromRoot := func() (err error) {
+		if cmd, ok := os.LookupEnv("SUDO_COMMAND"); ok && cmd != "" {
+			var uid, gid int
+			uid, err = strconv.Atoi(os.Getenv("SUDO_UID"))
+			if err != nil {
+				err = errors.Annotate(err, "parsing SUDO_UID")
+				return
+			}
+			gid, err = strconv.Atoi(os.Getenv("SUDO_GID"))
+			if err != nil {
+				err = errors.Annotate(err, "parsing SUDO_GID")
+				return
+			}
+			return syscall.Chown(flockName, uid, gid)
+		}
+		return nil
+	}
+	open := func() (fd int, err error) {
```

Comment:
Can we not use named return values and just have explicit returns? There are only two.

Reply:
Of course